lib/redux/actions/async.js

```diff
@@ -557,6 +625,17 @@ export function setUploadTargetUserAndMaybeRedirect(targetId) {
     });
     if (_.isEmpty(targetedDevices) || !atLeastOneDeviceSupportedOnSystem) {
       return dispatch(syncActions.setPage(pages.SETTINGS));
+    } else {
+      if (actionUtils.userHasRole(_.get(allUsers, loggedInUser), 'clinic')) {
```

Comment:
This is actually what got me to put that `if` in above. It is necessary here because we don't want to redirect users in the non-clinic case since they'll already be on the `MAIN` page when they're using the user dropdown, but clinic users are on the `CLINIC_USER_SELECT` page and need to be redirected.

I think you may be correct that this could be simplified by using a dedicated action creator for the clinic experience.